lib/logflare/alerting.ex

```diff
@@ -192,6 +191,33 @@ defmodule Logflare.Alerting do
     end)
   end
 
+  def sync_alert_jobs do
+    init_alert_jobs()
+    |> Enum.each(fn job ->
+      remote_call_scheduler(:add_job, [job])
+    end)
+  end
+
+  @doc """
+  Syncs a specific alert job by alert_id.
+  Upserts the job if it doesn't exist, otherwise deletes the existing job.
+  """
+  @spec sync_alert_job(number()) :: :ok | {:error, :not_found}
+  def sync_alert_job(alert_id) when is_integer(alert_id) do
+    alert_query = get_alert_query_by(id: alert_id)
+
+    if alert_query do
+      upsert_alert_job(alert_query)
+    else
+      # alert query does not exist, maybe remove from scheduler
+      job = get_alert_job(alert_id)
+
+      if job do
+        delete_alert_job(alert_id)
+      end
+    end
```

Comment:
This is doing multiple `:erpc` calls, passing data back and forth. For example, `get_alert_job` sends the whole job struct from the scheduler node to the caller, just to do another `:erpc` call, sending the alert id back.

Ideally you want a single `:erpc` call that does all the logic at once, on the scheduler node. In case of `sync_alert_job` we only need to send the `alert_id` to the scheduler node, and get `:ok | {:error, :not_found}` back, so there is little sent over the wire.

One way would be something like this:

```elixir
def sync_alert_job(alert_id) when is_integer(alert_id) do
  on_scheduler_node(fn ->
    # We can call AlertsScheduler here directly.
  end)
end
```

What do you think?